Tasks/DeployVisualStudioTestAgent/InstallTestAgent.ps1

```diff
@@ -83,20 +76,32 @@ function Install-Product($SetupPath, $ProductVersion, $Update)
 
 	try
 	{
+        # Invoke the TA installation
 		if($ProductVersion -eq "15.0") {
 			$tpPath = Join-Path $env:SystemDrive TestAgent2017
-			$argumentsarr = @("--noRestart","--quiet","--installPath","$tpPath")
+			$exitCode = InstallTestAgent2017 -SetupPath $SetupPath -InstallPath $tpPath
```

Comment:
>$exitCode [](start = 3, length = 9)

we r not checking for exit code before configuration. we might eat up exit code of installer and then it will be difficult to debug any issues